pkg/imgpkg/bundle/images_lock.go

```diff
@@ -93,9 +100,37 @@ func (o *ImagesLock) generateImagesLocations(relativeToRepo string) {
 	o.imageRefs = result
 }
 
-func (o ImagesLock) ImageRefs() ImageRefs {
-	return o.imageRefs
+func (o *ImagesLock) ImageRefs() (ImageRefs, error) {
+	err := o.syncImageRefs()
+	if err != nil {
+		return ImageRefs{}, err
+	}
+	return o.imageRefs, nil
+}
+
+func (o *ImagesLock) syncImageRefs() error {
+	locationsConfig, err := o.imagesLockLocationFetcher.Config()
+	if _, ok := err.(*LocationsNotFound); ok {
+		return nil
+	}
```

Comment:
i'm going to leave it as-is since it is a 'nit' and also to me a flatter structure is a lot easier to comprehend (and is the reason I decided to avoid nesting it in the first place)